game-core/src/main/java/games/strategy/engine/framework/startup/login/HmacSha512Authenticator.java

```diff
@@ -39,36 +39,31 @@
    *         node; never {@code null}.
    */
   static Map<String, String> newChallenge() {
-    // This length is relatively large for a nonce simply because we do not currently expire nonces after they have been
-    // issued. If this were done, and a short nonce lifetime was used, we could probably get away with 8 bytes of random
-    // data safely as long as we included the timestamp as part of the nonce.
-    //
-    // https://security.stackexchange.com/questions/1952/how-long-should-a-random-nonce-be
-    final int nonceLengthInBytes = 64;
-
-    // Salt length should match hash output size.
-    //
-    // https://crackstation.net/hashing-security.htm#salt
-    final int saltLengthInBytes = 64;
-
     return Maps.newHashMap(ImmutableMap.<String, String>builder()
-        .put(encodeProperty(ChallengePropertyNames.NONCE, newRandomBytes(nonceLengthInBytes)))
-        .put(encodeProperty(ChallengePropertyNames.SALT, newRandomBytes(saltLengthInBytes)))
```

Comment:
I'm not sure getting rid of the `length` parameter makes sense here.

`nonceLengthInBytes` and `saltLengthInBytes` have no relationship to each other.  They are completely orthogonal--it's simply coincidental that they're equal.  Someone coming along later will probably get confused and think there's a correlation between these two lengths when, in fact, salt length is actually correlated to the hash length.

The comment associated with `saltLengthInBytes` is equally important as the one for `nonceLengthInBytes`, especially if a future code reader is doing some kind of security audit.